src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -209,6 +209,17 @@ void GenericThreadStackManagerImpl_OpenThread<ImplClass>::_OnPlatformEvent(const
 #endif
         Impl()->LockThreadStack();
 
+        if (mInitialDeviceType == ConnectivityManager::kThreadDeviceType_SleepyEndDevice && event->ThreadStateChange.RoleChanged)
+        {
+            otLinkModeConfig linkMode = otThreadGetLinkMode(mOTInst);
+            linkMode.mRxOnWhenIdle    = (otThreadGetDeviceRole(mOTInst) !=
```

Comment:
So this switch the device back to MED when it loses connectivity with a Thread router. Is that expected? Not sure it's a good decision for the battery lifetime. I think we may need at least config switch for such behavior.